dspace-server-webapp/src/test/java/org/dspace/app/rest/utils/URLUtilsTest.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.utils;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Ignore;
+import org.junit.Test;
+
+/**
+ * @author mwood
+ */
+public class URLUtilsTest {
+
+    public URLUtilsTest() {
+    }
+
+    /**
+     * Test of decode method, of class URLUtils.
+     */
+    @Ignore
+    @Test
+    public void testDecode() {
+    }
+
+    /**
+     * Test of encode method, of class URLUtils.
+     */
+    @Ignore
+    @Test
+    public void testEncode() {
+    }
+
+    /**
+     * Test of urlIsPrefixOf method, of class URLUtils.
+     */
+    @Test(expected = IllegalArgumentException.class)
+    @SuppressWarnings("UnusedAssignment")
+    public void testUrlIsPrefixOf() {
+        boolean isPrefix;
+
+        isPrefix = URLUtils.urlIsPrefixOf("http://example.com/path", "http://example.com/path");
+        assertTrue("Should match if all is equal", isPrefix);
+        isPrefix = URLUtils.urlIsPrefixOf("http://example.com:80/test", "http://example.com:80/test/1");
+        assertTrue("Should match if pattern path is longer", isPrefix);
+        isPrefix = URLUtils.urlIsPrefixOf("http://example.com:80/test", "http://example.com/test");
+        assertTrue("Should match if missing port matches default", isPrefix);
+
+        isPrefix = URLUtils.urlIsPrefixOf("http://example.com/path", "https://example.com/path");
```

Comment:
can we add test for 
- http://example.com/path vs http://example.com/path/
- http://example.com/path vs http://example.com/path2
- http://example.com/path vs http://example.com/path2/sub

Reply:
Thank you.  Those are good suggestions.  Also, working on this revealed that I had the parameters reversed.

I took the opportunity to remove a "FIXME" by replacing the string comparison of paths with an element-by-element comparison which should be more accurate.